zeeguu_api/api/search.py

```diff
@@ -0,0 +1,182 @@
+
+
+# ---------------------------------------------------------------------------
+@api.route(f"/{SUBSCRIBE_SEARCH}/<search_terms>", methods=("GET",))
+# ---------------------------------------------------------------------------
+@cross_domain
+@with_session
+def subscribe_to_search(search_terms):
+    """
+    :param: search_terms -- the search terms to be subscribed to.
+    Subscribe to a search with given search terms.
+
+    :return: it returns the search as a dictionary. Like:
+                id = unique id of the search;
+                search_keywords = <unicode string>
+             This is used to display it in the UI.
+
+    """
+    search = Search.find_or_create(session, search_terms)
+    SearchSubscription.find_or_create(session, flask.g.user, search)
+
+    return json_result(search.as_dictionary())
+
+
+# ---------------------------------------------------------------------------
+@api.route(f"/{UNSUBSCRIBE_SEARCH}/<search_id>", methods=("GET",))
+# ---------------------------------------------------------------------------
+@cross_domain
+@with_session
+def unsubscribe_from_search(search_id):
+    """
+    A user can unsubscribe from the search with a given ID
+
+    :return: OK / ERROR
+    """
+    try:
+        to_delete = SearchSubscription.with_search_id(search_id, flask.g.user)
+        session.delete(to_delete)
+        to_delete2 = Search.find_by_id(search_id)
+        session.delete(to_delete2)
+        session.commit()
+
+    except Exception as e:
+        zeeguu.log(str(e))
+        return "OOPS. SEARCH AIN'T THERE IT SEEMS (" + str(e) + ")"
+
+    return "OK"
+
+
+# ---------------------------------------------------------------------------
+@api.route(f"/{SUBSCRIBED_SEARCHES}", methods=("GET",))
+# ---------------------------------------------------------------------------
+@cross_domain
+@with_session
+def get_subscribed_searches():
+    """
+    A user might be subscribed to multiple searches at once.
+    This endpoint returns them as a list.
+
+    :return: a json list with searches for which the user is registered;
+     every search in this list is a dictionary with the following info:
+                id = unique id of the search;
+                search_keywords = <unicode string>
+    """
+    subscriptions = SearchSubscription.all_for_user(flask.g.user)
+    searches_list = []
+
+    for subs in subscriptions:
+        try:
+            searches_list.append(subs.search.as_dictionary())
+        except Exception as e:
+            zeeguu.log(str(e))
+
+    return json_result(searches_list)
+
+
+# ---------------------------------------------------------------------------
+@api.route(f"/{FILTER_SEARCH}/<search_terms>", methods=("GET",))
+# ---------------------------------------------------------------------------
+@cross_domain
+@with_session
+def filter_search(search_terms):
+    """
+    :param: search_terms -- the search to be filtered.
+    Subscribe to the search filter with the given terms.
```

Comment:
what the function does should come before the :params:

Reply:
Changed it and added return.